docs/howtos/building-a-rust-component.md

```diff
@@ -1,20 +1,179 @@
+summaried as:
+
+* Every table which defines the "top level" items being synced will have a
+  column called something like 'sync_change_counter'
+
+* A trigger will be used so that every write to this table will increment this
+  change counter.
+
+* At sync time, items with a non-zero change counter are candidates for syncing.
+
+* As the sync starts, for each item, the current value of the change counter is
+  remembered. At the end of the sync, the counter is decremented by this value.
+  Thus, items which were changed between the time the sync started and completed
+  will be left with a non-zero change counter at the end of the sync.
+
+## Syncing FAQs
+
+This section is stolen from [this document](https://docs.google.com/document/d/1s9ld2F4e83eQ944kN6QXXTRlqrX74w2AJS6W2fDyAJ8)
+
+### What’s the global sync ID and the collection sync ID?
+Both guids, both used to identify when the data in the server has changed
+radically underneath us (eg, when looking at lastModified is no longer a sane
+thing to do.)
+
+The "global sync ID" changing means that every collection needs to be assumed as
+having changed radically, whereas just the "collection sync ID" changing means
+just that one collection.
+
+### What’s `get_sync_assoc`, why is it important? What is `StoreSyncAssociation`?
+They are all used to track the guids above. It’s vitally important we know when
+these guids change.
+
+StoreSyncAssociation is a simple enum which reflects the state a sync engine
+can be in - either `Disconnected` (ie, we have no idea what the GUIDs are) or
+`Connected` where we know what we think the IDs are (but the server may or may
+not match with this)
+
+These GUIDs will typically be stored in the DB in the metadata table.
+
+### what is `apply_incoming` versus `sync_finished`
+`apply_incoming` is where any records incoming from the server (ie, possibly
+all records on the server if this is a first-sync, records with a timestamp
+later than our last sync otherwise) are processed.
+
+`sync_finished` is where we've done all the sync work other than uploading new
+changes to the server.
+
+### What's the diff between reset and wipe?
+
+* Reset means “I don’t know what’s on the server - I need to reconcile everything there with everything I have”. IOW, a “first sync”
+* Wipe means literally “wipe all server data”
+
+### The "engine" concept is weird, I assumed I have to implement it but it's not really relevant anymore coz since manager?
+
+Yeah, we should clean this up - but in general, you just have to implement a
+trait or 2 - all the sync machinery is already handled.
+
+# Exposing to consumers
+
+You will need an FFI or some other way of exposing stuff to your consumers.
 
-Please share your :+1: on
-[the relevant github issue](https://github.com/mozilla/application-services/issues/598)
-to let us know that you wanted it.
+We are moving towards uniffi - but this is so new that it's not even clear yet
```

Comment:
Heh, "moving towards uniffi" sounds premature just yet, even to me; "tentatively exploring" maybe...